File: python-packages/contracts-py/src/contracts_py/receipt.py
"""Receipt acknowledgement Pydantic model."""

from __future__ import annotations

from datetime import datetime
from typing import Optional
from pydantic import BaseModel

from .enums import ReceiptStatus


class ReceiptAcknowledgement(BaseModel):
    """Acknowledgement of forecast message receipt."""

    schema_version: int
    message_id: str
    receipt_status: ReceiptStatus
    timestamp: datetime
    receipt_id: str
    detail: Optional[str] = None

File: crates/protocol/src/receipt.rs
//! Receipt acknowledgement schema.
//!
//! Sent by the Rust core to acknowledge receipt of a forecast message.

use crate::enums::ReceiptStatus;
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

/// Acknowledgement of forecast message receipt.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReceiptAcknowledgement {
    /// Schema version
    pub schema_version: u32,
    /// Reference to the original message
    pub message_id: String,
    /// Receipt status
    pub receipt_status: ReceiptStatus,
    /// When the acknowledgement was generated
    pub timestamp: DateTime<Utc>,
    /// Receipt sequence number
    pub receipt_id: String,
    /// Optional detail for rejections
    pub detail: Option<String>,
}
